### DISCUSSION TITLE
How do I obtain the value of the GO_EA_AUTO_REGISTER_ELASTIC_AGENT_ID environment variable in gocd-agent

### DISCUSSION THREAD

**Discussion by xlw712** [gocd/gocd#11536] [Q&A] [answered]
I want to get the correct pod name when the agent runs,because I set pod config hostNetwork:true, get hostname can only 
I tried a lot of things.
1st
![image](https://user-images.githubusercontent.com/10722207/236167594-7cd1ef60-5a2c-494c-a386-e2138adf68f9.png)
2nd
I suspect it's because of different shells,But if I go into the agent-pod, echo $GO_EA_AUTO_REGISTER_ELASTIC_AGENT_ID will get the correct pod name
![image](https://user-images.githubusercontent.com/10722207/236167814-3f33133a-39e2-40fd-9805-ca4f88f546fb.png)

**Comment by xlw712** [comment]
Is there any way to get it

**Reply by chadlwilson** [reply, marked as answer]
If you don't see it with your `env` print-out in the 2nd step, then no.

Not all environment variables are propagated from, say, docker container through to individual agent tasks spawned.
